public/javascripts/account-main.js

```diff
@@ -0,0 +1,35 @@
+function fetchUserHighScores() {
+  fetch(`fetch-user-highscores`, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      let htmlRows = [];
+      data.lengthSettingsOptions.forEach((setting, index) => {
+        htmlRows[index] = `<div class="grid-length">${setting}</div>`;
+      });
+      for (let i = 0; i < Object.keys(data.highscores).length; i++) {
+        for (let j = 0; j < data.lengthSettingsOptions.length; j++) {
+          if (!data.highscores[i + 1][data.lengthSettingsOptions[j]]) {
+            htmlRows[i] += `<div class="grid-score">0</div>`;
+          } else {
+            htmlRows[i] += `<div class="grid-score">${
+              data.highscores[i + 1][data.lengthSettingsOptions[j]]
+            }</div>`;
+          }
+        }
+      }
+      const highscoresTable = document.querySelector(".grid-container");
+      htmlRows.forEach((row) => {
```

Comment:
Why not append html to the document while iterating in the loops above? Also consider switching to ES6 methods, looping is not ideal

Reply:
Wouldnt appending constantly while looping be more costly performance-wise? Changed a bit how the `div`s are generated, so can add them all in one go now, which seems more efficient.

Reply:
That is a good point actually, yes you should keep it like that then.